### PR TITLE
Add GeoTessCPP recipe

### PR DESCRIPTION
**Pull request by jkmacc-LANL**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by jkmacc-LANL** [comment]
@conda-forge-admin, please ping conda-forge/staged-recipes

I'm having trouble with the source patch for this project.  It's hard to find how this is done in the docs, so it's hard to test this locally.  Is anyone able to help me decipher how to fix [this error](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=959947&view=logs&j=e35d4f76-8ff2-5536-d795-df91e63eb9f7&t=fa7b4b17-b6ff-5c9c-8cfc-15f888c92310&l=1330) in the build?

```
Downloading https://github.com/sandialabs/GeoTessCPP/archive/v2.7.2.tar.gz
Success
Extracting download
Applying patch: /Users/runner/work/1/s/recipes/GeoTessCPP/0001-Don-t-overwrite-CC-for-conda-forge.patch
Applying patch: /Users/runner/work/1/s/recipes/GeoTessCPP/0001-Don-t-overwrite-CC-for-conda-forge.patch with args:
['-Np1', '-i', '/var/folders/4z/th_0342d7sv9wl7kb3yfllq80000gn/T/tmpqi5l778p/0001-Don-t-overwrite-CC-for-conda-forge.patch.native', '--binary']
can't find file to patch at input line 20
Perhaps you used the wrong -p or --strip option?
...
```

Thank you!

**Comment by jkmacc-LANL** [comment]
ping @conda-forge/help-c-cpp

**Comment by jkmacc-LANL** [comment]
I've tried to skip `[win]` while I figure it out locally, so I think this is ready to go on my end.  Thank you!

File: recipes/GeoTessCPP/LICENSE.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-Copyright (c) 2015-2023, conda-forge
```
Comment (ocefpaf) [inline recipes/GeoTessCPP/LICENSE.txt:1]: This is the staged-recipes license. Please don't change it or use it as the package license. Please revert this change.

File: recipes/GeoTessCPP/build.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+
+set -ex
+
+git config --global --add safe.directory /home/conda/staged-recipes
```
Comment (ocefpaf) [inline recipes/GeoTessCPP/build.sh:5]: Is this really needed?

File: recipes/GeoTessCPP/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+about:
+  home: https://www.sandia.gov/geotess/ 
+  summary: '3D Earth model parameterization, storage, and interrogation software'
+  description: |
+    A model parameterization and software support system that implements the
+    construction, population, storage and interrogation of data stored in 3D Earth
+    models. This version is in C++ and intended for Linux systems.
+  license: BSD-3-Clause
+  license_family: BSD
+  license_file: LICENSE.txt
```
Comment (ocefpaf) [inline recipes/GeoTessCPP/meta.yaml:36]: This is this package license: https://github.com/sandialabs/GeoTessCPP/blob/master/LICENSE

File: recipes/GeoTessCPP/LICENSE.txt (hunk 1 above)
Comment (jkmacc-LANL) [reply recipes/GeoTessCPP/LICENSE.txt:1]: Done, my apologies.  Misunderstood the docs.

File: recipes/GeoTessCPP/build.sh (hunk 2 above)
Comment (jkmacc-LANL) [reply recipes/GeoTessCPP/build.sh:5]: Thought I needed it for local Linux build to finish.  Removing.

**Comment by jkmacc-LANL** [comment]
I think I've done as requested.  Thank you.

File: LICENSE.txt [hunk 4]
```diff
@@ -1,27 +0,0 @@
-Copyright (c) 2015-2023, conda-forge
```
Comment (ocefpaf) [inline LICENSE.txt:1]: Please don't remove our license!

File: recipes/GeoTessCPP/build.sh [hunk 5]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+
+set -e
+
+make all CC=${CC}
```
Comment (ocefpaf) [inline recipes/GeoTessCPP/build.sh:5]: Is there a `make check` we can run here?

Comment (jkmacc-LANL) [reply recipes/GeoTessCPP/build.sh:5]: No, there doesn't seem to be.

File: LICENSE.txt (hunk 4 above)
Comment (jkmacc-LANL) [reply LICENSE.txt]: I copied it back in.  My git-foo isn't good enough to figure out how to "never have removed it."  Want me to squash this PR into one commit?

Comment (ocefpaf) [reply LICENSE.txt]: Sure. That works.

**Comment by jkmacc-LANL** [comment]
I don't know why the license looks like the last line is changed, but I'm happy with everything else.  Thank you, again.

**Comment by ocefpaf** [comment]
> I don't know why the license looks like the last line is changed, but I'm happy with everything else. Thank you, again.

I just fixed that.
